pmm-app/tests/qan_test.js

```diff
@@ -148,15 +150,103 @@ xScenario('Open the QAN Dashboard and check Details -> Tables', async (I, adminP
 
 Feature('To verify and test the QAN Dashboard query details tooltip');
 
-Before((I, loginPage) => {
-  I.amOnPage(loginPage.url);
-  loginPage.login('admin', 'admin');
+Before((I, qanPage) => {
+  I.Authorize();
+  I.amOnPage(qanPage.url);
 });
 
-xScenario('Open the QAN Dashboard and show tooltip', async (I, adminPage, qanPage) => {
+xScenario('Open the QAN Dashboard and show tooltip @new-qan', async (I, adminPage, qanPage) => {
   I.amOnPage(qanPage.url);
   qanPage.showTooltip(4, 1);
   I.wait(5);
   I.see('dasdadadad');
   // TODO: check changes in url
 });
+
+Scenario('Open the QAN Dashboard and check existence of filters @new-qan', async (I, qanPage) => {
+  await qanPage.verifyFiltersSectionIsPresent();
+});
+
+Scenario('Open the QAN Dashboard and check work of button "reset all" @new-qan', async (I, qanPage) => {
+  qanPage.selectFilter(qanPage.fields.filterCheckboxSelector);
+  qanPage.resetAllFilters();
+  I.dontSeeCheckboxIsChecked(qanPage.fields.filterCheckboxSelector);
+});
+
+Scenario('Open the QAN Dashboard, add a column and make sure that this option not available anymore @new-qan', async (I, qanPage) => {
+  const COLUMN_NAME = 'Bytes Sent';
+
+  qanPage.addColumn(COLUMN_NAME);
+  await qanPage.verifyColumnIsNotAvailable(COLUMN_NAME);
+});
+
+Scenario('Open the QAN Dashboard and verify that hovering over a time metric displays a tooltip with a graph @new-qan', async (I, qanPage) => {
+  const ROW_NUMBER = 1;
+  const QUERY_TIME_COLUMN_NUMBER = 3;
+
+  qanPage.showTooltip(ROW_NUMBER, QUERY_TIME_COLUMN_NUMBER);
+  I.seeElement(qanPage.elements.latencyChart);
+});
+
+Scenario('Open the QAN Dashboard and verify that hovering over a non-time metric displays a tooltip without a graph @new-qan', async (I, qanPage) => {
+  const ROW_NUMBER = 1;
+  const QUERY_COUNT_COLUMN_NUMBER = 2;
+
+  qanPage.showTooltip(ROW_NUMBER, QUERY_COUNT_COLUMN_NUMBER);
+  I.dontSeeElement(qanPage.elements.latencyChart);
+});
+
+Scenario('Open the QAN Dashboard and verify that the metric value matches the "Per sec" value in the tooltip. @new-qan', async (I, qanPage) => {
+  const ROW_NUMBER = 1;
+  const QUERY_COUNT_COLUMN_NUMBER = 2;
+
+  qanPage.verifyMetricsMatch(ROW_NUMBER, QUERY_COUNT_COLUMN_NUMBER);
```

Comment:
await needed for verifyMetricsMatch method
  please change method verifyMetricsMatch
  i expect test to look like :
  const ROW_NUMBER
  const QUERY_COUNT_COLUMN_NUMBER
  const cellSelector = this.overviewMetricCellValueLocator(ROW_NUMBER, QUERY_COUNT_COLUMN_NUMBER);
  showTooltip(ROW_NUMBER, QUERY_COUNT_COLUMN_NUMBER)
  verifyMetricsMatch(cellSelector, qaPage.qpsTooltipValueSelector) - this way we will have reusable assertion

Reply:
I believe it will be a little overkill to do it this way if we want to make tests high level we may hide this behavior to qanPage, and there is no to much sense to have reusable assertion because this check only will work after tooltip is opened. So code is strongly connected

Reply:
as i see assertion compares 2 values from given selectors. we can compare not only tooltip , but details and overview also, or any other raw number values

Reply:
Not exactly, it's not for raw number values, but for values that contains text inside elements. We shouldn't create a separate method for it because now it's single case when we met it. I'd propose to not overload qanPage with small helpers and do it if we really need it, for example, if we'll have more checks  for not only numeric values

Reply:
Anyway, I'm going to work on our tests refactoring after we'll have little more tests rewritten to new QAN and also I'll add some helpers, including this one